src/coprocessor/dag/dag.rs

```diff
@@ -63,16 +66,26 @@ impl DAGContext {
             exec: dag_executor.exec,
             output_offsets: req.take_output_offsets(),
             batch_row_limit: batch_row_limit,
+            deadline: deadline,
         })
     }
 
+    #[inline]
+    fn check_outdated(&self) -> Result<()> {
```

Comment:
I think it's not a good idea to check_outdated in `DagContext` since almost all requests in coprocessor needs this function.

Reply:
There is another `check_outdated` in `RequestTask`.